src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -389,6 +391,67 @@ public class SystemNamespaceSymbol : NamespaceSymbol
                 .WithReturnType(LanguageConstants.String)
                 .WithDescription("Returns a value in the format of a globally unique identifier. **This function can only be used in the default value for a parameter**.")
                 .WithFlags(FunctionFlags.ParamDefaultsOnly)
+                .Build(),
+
+            new FunctionOverloadBuilder("loadContent")
+                .WithReturnType(LanguageConstants.String)
+                .WithDescription("Load content of specified file into a string. Content loading occurs during compilation, not in runtime. Maximum allowed content size is 24,576 characters (including line endings).")
+                .WithRequiredParameter("filePath", LanguageConstants.String, "File path which content will be loaded")
+                .WithOptionalParameter("encoding", LanguageConstants.String, "File encoding. If not provided, Bicep will try to automatically detect the encoding of a file based on the presence of byte order marks.")
+                .WithDiagnosticsCollector((binder, diagnostics, arguments) => {
```

Comment:
>WithDiagnosticsCollector [](start = 17, length = 24)

This is nice!

Reply:
I thought I can have something similar for writing template, but not sure if that will be possible, FunctionOverload is available only in TypeAssignmentVisitor and after assigning type it's gone, I'd need to make the type to carry on the used FunctionOverload or just the delegate...